contracts/otcMaker/OtcMaker.sol

```diff
@@ -38,33 +41,61 @@ contract OtcMaker is SafeOwnable, EIP712Upgradeable, IOtcMaker, OtcMakerStorageV
         _;
     }
 
+    modifier onlyFundOwner() {
+        // OM_NFO: not fund owner
+        require(_msgSender() == _fundOwner, "OM_NFO");
+        _;
+    }
+
+    modifier onlyPositionManager() {
+        // OM_NPM: not position manager
+        require(_msgSender() == _positionManager, "OM_NPM");
+        _;
+    }
+
     //
     // EXTERNAL NON-VIEW
     //
 
     function initialize(address clearingHouseArg, address limitOrderBookArg) external initializer {
         __SafeOwnable_init();
         _caller = _msgSender();
+        _fundOwner = _msgSender();
+        _positionManager = _msgSender();
         _clearingHouse = clearingHouseArg;
         _limitOrderBook = limitOrderBookArg;
         _vault = IClearingHouse(_clearingHouse).getVault();
         _accountBalance = IClearingHouse(_clearingHouse).getAccountBalance();
     }
 
-    function setCaller(address newCaller) external override onlyOwner {
-        // OM_ZA: zero address
-        require(newCaller != address(0), "OM_ZA");
+    function setCaller(address newCaller) external onlyOwner {
+        _requireNonZeroAddress(newCaller);
+        address oldCaller = _caller;
         _caller = newCaller;
-        emit UpdateCaller(_caller, newCaller);
+        emit CallerUpdated(oldCaller, newCaller);
+    }
+
+    function setFundOwner(address newFundOwner) external onlyOwner {
+        _requireNonZeroAddress(newFundOwner);
+        address oldFundOwner = _fundOwner;
+        _fundOwner = newFundOwner;
+        emit FundOwnerUpdated(oldFundOwner, newFundOwner);
+    }
+
+    function setPositionManager(address newPositionManager) external onlyOwner {
+        _requireNonZeroAddress(newPositionManager);
+        address oldPositionManager = _positionManager;
+        _positionManager = newPositionManager;
+        emit PositionManagerUpdated(oldPositionManager, newPositionManager);
     }
 
-    function setMarginRatioLimit(uint24 marginRatioLimitArg) external override onlyOwner {
+    function setMarginRatioLimit(uint24 marginRatioLimitArg) external onlyOwner {
         require(marginRatioLimitArg > 62500 && marginRatioLimitArg < 1000000, "OM_IMR");
```

Comment:
To prevent hardcode `62500`, suggest to query this value from ClearingHouseConfig.
```
uint24 mmRatio = IClearingHouseConfig(IClearingHouse(_clearingHouse).getClearingHouseConfig()).getMmRatio();

require(marginRatioLimitArg > mmRatio && marginRatioLimitArg < 1000000, "OM_IMR");
```